contracts/mocks/CloneRewarderTimeDual.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+import "../interfaces/IRewarder.sol";
+import "@boringcrypto/boring-solidity/contracts/libraries/BoringERC20.sol";
+import "@boringcrypto/boring-solidity/contracts/libraries/BoringMath.sol";
+import "@boringcrypto/boring-solidity/contracts/BoringOwnable.sol";
+
+interface IMasterChefV2 {
+    function lpToken(uint256 pid) external view returns (IERC20 _lpToken); 
+}
+
+/// @author @0xKeno
+contract CloneRewarderTimeDual is IRewarder,  BoringOwnable{
+    using BoringMath for uint256;
+    using BoringMath128 for uint128;
+    using BoringERC20 for IERC20;
+
+    IERC20 public rewardToken1;
+    IERC20 public rewardToken2;
+
+    /// @notice Info of each MCV2 user.
+    /// `amount` LP token amount the user has provided.
+    /// `rewardDebt` The amount of SUSHI entitled to the user.
+    struct UserInfo {
+        uint256 amount;
+        uint256 rewardDebt1;
+        uint256 rewardDebt2;
+    }
+
+    /// @notice Info of each MCV2 pool.
+    /// `allocPoint` The amount of allocation points assigned to the pool.
+    /// Also known as the amount of SUSHI to distribute per block.
+    struct PoolInfo {
+        uint128 accToken1PerShare;
+        uint128 accToken2PerShare;
+        uint64 lastRewardTime;
```

Comment:
Can save one slot if we use uint112, uint112, uint32.

Reply:
Do you think the precision is still high enough?